pkg/git/repository.go

```diff
@@ -36,14 +37,15 @@ type (
 	}
 
 	CloneOptions struct {
-		// URL clone url
-		Repo     string
-		Auth     Auth
-		FS       fs.FS
-		Progress io.Writer
-		url      string
-		revision string
-		path     string
+		AutoCreate bool
```

Comment:
Lets call it `CreateIfNotExist` and add a comment above it so it's visible in our `godoc`s.
`// CreateIfNotExist will try to create the repository if it does not exist`.